### PR TITLE
fix: Add missing parameters

### PR DESCRIPTION
**Pull request by sfc-gh-jcieslak**
A fix for #2248

### CODE REVIEW

File: pkg/resources/grant_privileges_to_role.go [hunk 1]
```diff
@@ -65,6 +65,7 @@ var grantPrivilegesToRoleSchema = map[string]*schema.Schema{
 						"INTEGRATION",
 						"FAILOVER GROUP",
 						"REPLICATION GROUP",
+						"EXTERNAL VOLUME",
```
Comment (sfc-gh-asawicki) [inline pkg/resources/grant_privileges_to_role.go:68]: You did not change the description. And they are also listed there. You will also have to regenerate docs after changing the description.

File: pkg/resources/grant_privileges_to_role.go [hunk 2]
```diff
@@ -191,6 +192,7 @@ var grantPrivilegesToRoleSchema = map[string]*schema.Schema{
 									"TASKS",
 									"VIEWS",
 									"MATERIALIZED VIEWS",
+									"ICEBERG TABLES",
```
Comment (sfc-gh-asawicki) [inline pkg/resources/grant_privileges_to_role.go:195]: same with description here

File: pkg/resources/grant_privileges_to_role.go [hunk 3]
```diff
@@ -245,6 +247,7 @@ var grantPrivilegesToRoleSchema = map[string]*schema.Schema{
...
```
Comment (sfc-gh-asawicki) [inline pkg/resources/grant_privileges_to_role.go:250]: same with descritpion here

File: pkg/resources/grant_privileges_to_role_acceptance_test.go [hunk 4]
```diff
@@ -2,6 +2,8 @@ package resources_test
 
 import (
 	"fmt"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+	"github.com/hashicorp/terraform-plugin-testing/tfversion"
```
Comment (sfc-gh-asawicki) [inline pkg/resources/grant_privileges_to_role_acceptance_test.go:6]: move imports pls

File: pkg/sdk/grants_test.go [hunk 5]
```diff
@@ -35,6 +35,22 @@ func TestGrantPrivilegesToAccountRole(t *testing.T) {
...
+				AllPrivileges: Bool(true),
+			},
+			on: &AccountRoleGrantOn{
+				AccountObject: &GrantOnAccountObject{
+					ExternalVolume: Pointer(NewAccountObjectIdentifier("ex volume")),
+				},
+			},
+			accountRole: NewAccountObjectIdentifier("role1"),
+		}
+		assertOptsValidAndSQLEquals(t, opts, `GRANT ALL PRIVILEGES ON EXTERNAL VOLUME "ex volume" TO ROLE "role1"`)
```
Comment (sfc-gh-asawicki) [inline pkg/sdk/grants_test.go:51]: I wonder if external volume identifier follows the normal identifier rules

File: pkg/sdk/privileges.go [hunk 6]
```diff
@@ -71,6 +72,9 @@ const (
 	AccountObjectPrivilegeMonitor            AccountObjectPrivilege = "MONITOR"
 	AccountObjectPrivilegeUsage              AccountObjectPrivilege = "USAGE"
 
+	// -- For EXTERNAL VOLUME
+	// AccountObjectPrivilegeUsage              AccountObjectPrivilege = "USAGE" (duplicate)
```
Comment (sfc-gh-asawicki) [inline pkg/sdk/privileges.go:76]: why this is commented out?

File: pkg/sdk/privileges.go [hunk 7]
```diff
@@ -178,6 +184,15 @@ const (
 	// USAGE [ , ... ]
 	SchemaObjectPrivilegeUsage SchemaObjectPrivilege = "USAGE"
 
+	// -- For ICEBERG TABLE
+	SchemaObjectPrivilegeApplyBudget SchemaObjectPrivilege = "APPLYBUDGET"
+	//SchemaObjectPrivilegeDelete      SchemaObjectPrivilege = "DELETE" (duplicate)
```
Comment (sfc-gh-asawicki) [inline pkg/sdk/privileges.go:189]: Why these are commented out?

**Review by sfc-gh-asawicki — CHANGES_REQUESTED** [review]

File: pkg/sdk/grants_test.go (hunk 5 above)
Comment (sfc-gh-jcieslak) [reply pkg/sdk/grants_test.go:51]: From the documentation it seems like a normal account object identifier, plus `An external volume is a named, account-level Snowflake object` - [docs](https://docs.snowflake.com/en/user-guide/tables-iceberg#external-volumes)

File: pkg/sdk/privileges.go (hunk 6 above)
Comment (sfc-gh-jcieslak) [reply pkg/sdk/privileges.go:76]: So it seems like we're adding the privileges as they appear in the documentation, e.g. for External Volume ... For Tables ... etc. For some of them, they can duplicate (e.g. tables and iceberg tables may have "SELECT") and the practice in this file was to comment the duplicate with the comment `(duplicate)` and the end.

File: pkg/sdk/privileges.go (hunk 7 above)
Comment (sfc-gh-jcieslak) [reply pkg/sdk/privileges.go:189]: same reason as the above

File: pkg/sdk/privileges.go (hunk 6 above)
Comment (sfc-gh-asawicki) [reply pkg/sdk/privileges.go:76]: Oh, okay. We should deal with that. Maybe as a subtask in redesign grants? (close after reading and adding it there if you agree)

File: pkg/resources/grant_privileges_to_role_acceptance_test.go [hunk 8]
```diff
@@ -861,3 +864,38 @@ func TestAcc_GrantPrivilegesToRole_onSchemaObject_futureInDatabase_externalTable
 		},
 	})
 }
+
+func TestAcc_GrantPrivilegesToRole_onSchemaObject_futureIcebergTables(t *testing.T) {
```
Comment (sfc-gh-asawicki) [inline pkg/resources/grant_privileges_to_role_acceptance_test.go:868]: +1 for that test

**Review by sfc-gh-asawicki — APPROVED** [review]

**Pull request merged**